src/main/java/net/dv8tion/jda/core/entities/Guild.java

```diff
@@ -732,6 +732,58 @@ default VoiceChannel getVoiceChannelById(long id)
      */
     SnowflakeCacheView<VoiceChannel> getVoiceChannelCache();
 
+    /**
+     * Populated list of {@link net.dv8tion.jda.core.entities.Channel channels} for this guild.
+     * This includes all types of channels, such as category/voice/text.
+     * <br>This includes hidden channels by default.
+     *
+     * <p>The returned list is ordered in the same fashion as it would be by the official discord client.
+     * <ol>
+     *     <li>TextChannel without parent</li>
+     *     <li>VoiceChannel without parent</li>
+     *     <li>Categories
+     *         <ol>
+     *             <li>TextChannel with category as parent</li>
+     *             <li>VoiceChannel with category as parent</li>
+     *         </ol>
+     *     </li>
+     * </ol>
+     *
+     * @return Immutable list of channels for this guild
+     *
+     * @see    #getChannels(boolean)
+     */
+    default List<Channel> getChannels()
+    {
+        return getChannels(true);
+    }
+
+    /**
+     * Populated list of {@link net.dv8tion.jda.core.entities.Channel channels} for this guild.
+     * This includes all types of channels, such as category/voice/text.
+     *
+     * <p>The returned list is ordered in the same fashion as it would be by the official discord client.
+     * <ol>
+     *     <li>TextChannel without parent</li>
+     *     <li>VoiceChannel without parent</li>
+     *     <li>Categories
+     *         <ol>
+     *             <li>TextChannel with category as parent</li>
+     *             <li>VoiceChannel with category as parent</li>
+     *         </ol>
+     *     </li>
+     * </ol>
+     *
+     *
+     * @param  includeHidden
+     *         Whether to include channels with denied {@link Permission#VIEW_CHANNEL View Channel Permission}
+     *
+     * @return Immutable list of channels for this guild
+     *
+     * @see    #getChannels()
+     */
+    List<Channel> getChannels(boolean includeHidden);
```

Comment:
Would this be better as separate method `#getVisibleChannels()`?

Reply:
I think it would separate the functionality unnecessarily.
A boolean here is completely appropriate.

Reply:
```java
List<Channel> channels;
if (hidden)
     channels = guild.getChannels();
else
     channels = guild.getVisibleChannels();
```

compared to

```java
List<Channels> channels = guild.getChannels(hidden);
```